app/src/main/java/com/cmput301f20t21/bookfriends/ui/request/MapDialog.java

```diff
@@ -0,0 +1,127 @@
+package com.cmput301f20t21.bookfriends.ui.request;
+
+import android.app.Dialog;
+import android.location.Address;
+import android.location.Geocoder;
+import android.os.Bundle;
+import android.view.KeyEvent;
+import android.view.View;
+import android.view.Window;
+import android.view.WindowManager;
+import android.view.inputmethod.EditorInfo;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+import com.cmput301f20t21.bookfriends.R;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.MapView;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MapDialog extends DialogFragment {
+    private GoogleMap myMap;
+    private EditText searchText;
+    private Button cancelSearchButton;
+    private Button confirmSearchButton;
+    private RequestViewModel vm;
+    private int position;
+
+    // constructor
+    public MapDialog(RequestViewModel vm, int position) {
+        this.vm = vm;
+        this.position = position;
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        Dialog dialog =  super.onCreateDialog(savedInstanceState);
+        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
+        dialog.setContentView(R.layout.map_dialog);
+
+        MapView mapView = (MapView) dialog.findViewById(R.id.map_view);
+
+        mapView.onCreate(dialog.onSaveInstanceState());
+        mapView.onResume();
+        mapView.getMapAsync(new OnMapReadyCallback() {
+            @Override
+            public void onMapReady(GoogleMap googleMap) {
+                LatLng edmontonLatLng = new LatLng(53.544388, -113.490929);
+                googleMap.moveCamera(CameraUpdateFactory.newLatLngZoom(edmontonLatLng, 12f));
+                myMap = googleMap;
```

Comment:
this might be a question we need to discuss, can we ask for permission to get the user's location and set the default location to the user location we got. If the permission is denied then we can set to a default location like Edmonton

It seems bad to make the default always Edmonton when a user could live in somewhere far like Asia

Reply:
Yes I can get the user's location but I think it's not in the requirement so I didn't do that. 
Will be implementing it now

Reply:
would it work if we do that in another PR? imo this PR meets the bare minimum for views already.

Reply:
Yeah sure. Another PRs for getting device location and as well as auto suggestions.